### PR TITLE
Document environment variables

### PR DESCRIPTION
**Pull request by paolobasso99**
1. Add a table of possible envirorment variables
2. Add env vars to docker-compose.yml
3. Add env vars to app.json

### CODE REVIEW

**Comment by benbusby** [comment]
Thanks for contributing, this is something that certainly needed more documentation. I'll do a full review this week, but before I do that, would you mind updating any spelling of "envirorment" to "environment"? I did a quick glance and noticed a few misspellings.

File: README.md [hunk 1]
```diff
@@ -49,6 +49,20 @@ If using Heroku Quick Deploy, **you can skip this section**.
...
+| ------------------ | -------------------------------------------------------------- |
+| WHOOGLE_USER       | The username for basic auth. WHOOGLE_PASS should also be used. |
+| WHOOGLE_PASS       | The password for basic auth. WHOOGLE_USER should also be used. |
+| WHOOGLE_PROXY_USER | The username of the proxy server.                              |
+| WHOOGLE_PROXY_PASS | The password of the proxy server.                              |
+| WHOOGLE_PROXY_TYPE | The type of the proxy server. For example "socks5".           |
+| WHOOGLE_PROXY_LOC  | The location of the proxy server (host or ip).                 |
+| EXPOSE_PORT        | The port where Whoogle will be exposed.                        |
+| HTTPS_ONLY         | Enforce HTTPS. Se the setion below.                            |
+
```
Comment (benbusby) [inline README.md:65]: A couple of spelling fixes and rewording here. Basic auth requires both user/pass to be set, so fixed the wording there as well to make that a bit more clear.

```suggestion
## Environment Variables
There are a few optional environment variables available for customizing a Whoogle instance:

| Variable           | Description                                                    |
| ------------------ | -------------------------------------------------------------- |
| WHOOGLE_USER       | The username for basic auth. WHOOGLE_PASS must also be set if used. |
| WHOOGLE_PASS       | The password for basic auth. WHOOGLE_USER must also be set if used. |
| WHOOGLE_PROXY_USER | The username of the proxy server.                              |
| WHOOGLE_PROXY_PASS | The password of the proxy server.                              |
| WHOOGLE_PROXY_TYPE | The type of the proxy server. Can be "socks5", "socks4", or "http".           |
| WHOOGLE_PROXY_LOC  | The location of the proxy server (host or ip).                 |
| EXPOSE_PORT        | The port where Whoogle will be exposed.                        |
| HTTPS_ONLY         | Enforce HTTPS. (See [here](https://github.com/benbusby/whoogle-search#https-enforcement))                            |

```

File: README.md [hunk 2]
```diff
@@ -167,6 +181,19 @@ docker build --tag whoogle-search:1.0 .
 docker run --publish 5000:5000 --detach --name whoogle-search whoogle-search:1.0
 ```
 
+You may want to set the environment:
```
Comment (benbusby) [inline README.md:194]:
```suggestion
Optionally, you can also enable some of the following environment variables to further customize your instance:
```

File: README.md [hunk 3]
```diff
@@ -257,7 +285,8 @@ Only needed if your setup requires Flask to redirect to HTTPS on its own -- gene
 Note: You should have your own domain name and [an https certificate](https://letsencrypt.org/getting-started/) in order for this to work properly.
 
 - Heroku: Ensure that the `Root URL` configuration on the home page begins with `https://` and not `http://`
-- Docker: Add `--build-arg use_https=1` to your run command
+- Docker build: Add `--build-arg use_https=1` to your run command
+- Docker image: st the environment variable HTTPS_ONLY=1
```
Comment (benbusby) [inline README.md:299]:
```suggestion
- Docker image: Set the environment variable HTTPS_ONLY=1
```

File: app.json [hunk 4]
```diff
@@ -1,8 +1,49 @@
...
+    "docker",
+    "heroku",
+    "adblock",
+    "degoogle",
+    "privacy"
+  ],
+  "stack": "container",
+  "env": {
+    "WHOOGLE_USER": {
+      "description": "The username for basic auth. WHOOGLE_PASS should also be used. Leave empty to disable.",
```
Comment (benbusby) [inline app.json:19]:
```suggestion
      "description": "The username for basic auth. WHOOGLE_PASS must also be set if used. Leave empty to disable.",
```

File: app.json [hunk 5]
```diff
@@ -1,8 +1,49 @@
...
+      "value": "",
+      "required": false
+    },
+    "WHOOGLE_PASS": {
+      "description": "The password for basic auth. WHOOGLE_USER should also be used. Leave empty to disable.",
```
Comment (benbusby) [inline app.json:24]:
```suggestion
      "description": "The password for basic auth. WHOOGLE_USER must also be set if used. Leave empty to disable.",
```

File: README.md [hunk 6]
```diff
@@ -49,6 +49,20 @@ If using Heroku Quick Deploy, **you can skip this section**.
   - Ubuntu: `sudo apt-get install -y libcurl4-openssl-dev libssl-dev`
   - Arch: `pacman -S curl openssl`
 
+## Environment variables
```
Comment (benbusby) [inline README.md:52]:
```suggestion
## Environment Variables
```

**Review by benbusby — CHANGES_REQUESTED** [review]
Looks good, just a few changes needed before merging.

**Comment by paolobasso99** [comment]
I have just applied the suggestions. I have also documented how to set the env vars for systemd.

**Pull request merged**